File: 0747-largest-number-at-least-twice-of-others/0747-largest-number-at-least-twice-of-others.cpp
class Solution {
public:
    int dominantIndex(vector<int>& nums) {
        vector<int> arr = nums;
       sort(nums.begin(),nums.end());
        int n=nums.size();
        int last = nums[n-1];
        int last1 = nums[n-2];
        if(last < (2*last1)){
            return -1;
        }
        
        int x;
        for (int i = 0; i < arr.size(); i++) {
            if (arr[i] == last) {
                x = i; 
            }
        }
        return x;
    
    }
};

File: 0747-largest-number-at-least-twice-of-others/0747-largest-number-at-least-twice-of-others.java
class Solution {
    public int dominantIndex(int[] nums) {
        
        int uniqueEle = 0;
        int index = -1;
        for(int i=0; i<nums.length; i++){
            if(nums[i]>uniqueEle){
                uniqueEle = nums[i];
                    index = i;
            
            }
        }
        
        for(int j=0; j<nums.length; j++){
            if(j != index){
                if(nums[j]*2 > uniqueEle)
                    return -1;
            }
        }
        return index;
    }
}